### PR TITLE
feat: polish instrumentation spike

### PR DESCRIPTION
**Pull request by JakubBajorekZ**

### CODE REVIEW

File: src/types/instrumentation_events.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  country: string;
+  githubId?: string;
+  isThirteenYo?: boolean;
+  parentsEmail?: string;
+  validationStatus?: string;
+  validationSource?: string;
+  accountType?: AccountType;
+};
+
+//export type NrStudentsEditionCreateStudent = {
```
Comment (JakubKotkowiak) [inline src/types/instrumentation_events.ts:29]: What about those commented lines?

File: src/utils/distributed_tracing.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import Newrelic from 'newrelic';
+import { LambdaResponse } from '../types/response';
+
+type CallbackFunction = () => Promise<LambdaResponse>;
+
+export const handleDistributedTracing = async (traceName: string, f: CallbackFunction): Promise<LambdaResponse> => {
```
Comment (JakubKotkowiak) [inline src/utils/distributed_tracing.ts:6]: If possible, please rename the f: to function: to increase clarity

File: src/utils/distributed_tracing.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  let res;
+  await Newrelic.startWebTransaction(traceName, async function() {
```
Comment (JakubKotkowiak) [inline src/utils/distributed_tracing.ts:8]: This section looks a bit odd

**Review by JakubKotkowiak — CHANGES_REQUESTED** [review]

File: src/types/instrumentation_events.ts (hunk 1 above)
Comment (JakubBajorekZ) [reply src/types/instrumentation_events.ts:29]: Its placeholder for future full implementation, i think we can leave it there.

File: src/utils/distributed_tracing.ts (hunk 2 above)
Comment (JakubBajorekZ) [reply src/utils/distributed_tracing.ts]: fixed

File: src/functions/save.ts [hunk 4]
```diff
@@ -6,56 +6,67 @@ import { StudentDTO, studentDTOSchema } from '../types/person';
...
+    Newrelic.recordCustomEvent('NrStudentsEditionSaveData', {...nrEvent, ...{ action: 'success' }});
+    return {
+      headers: {
+        'Access-Control-Allow-Origin': '*'
+      },
+      statusCode: StatusCode.CREATED,
+      body: ''
+    };
+  });
+};
```
[the remark this answers is not in the document]
Comment (JakubBajorekZ) [reply src/functions/save.ts]: fixed

File: src/utils/distributed_tracing.ts (hunk 3 above)
Comment (JakubBajorekZ) [reply src/utils/distributed_tracing.ts:8]: We discussed it and i we decided its ok for now

**Review by JakubKotkowiak — APPROVED** [review]
Aside from the commented section, it looks good

**Review by JakubKotkowiak — APPROVED** [review]

**Pull request merged**